ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/spanning-tree.j2

```diff
@@ -9,6 +9,43 @@ spanning-tree mode {{ spanning_tree.mode }}
 no spanning-tree vlan-id {{ spanning_tree.no_spanning_tree_vlan }}
 {%     endif %}
 {%     if spanning_tree.mode == "mstp" %}
-spanning-tree mst 0 priority {{ spanning_tree.priority }}
+{%         if spanning_tree.mst_instances is defined and spanning_tree.mst_instances is not none  %}
+{%             for mst_instance in spanning_tree.mst_instances | arista.avd.natural_sort %}
+spanning-tree mst {{ mst_instance }} priority {% if spanning_tree.mst_instances[mst_instance].priority is defined and spanning_tree.mst_instances[mst_instance].priority is not none %}{{ spanning_tree.mst_instances[mst_instance].priority }}
```

Comment:
You will need to put the `if` before the config line. Otherwise you may end up rendering `spanning-tree mst 1 priority` if the priority keyword is missing.

Reply:
Very good point, fixed it.

Reply:
Good point, fixed!